arch/arm/core/cortex_m/fault.c

```diff
@@ -1114,7 +1066,23 @@ void z_arm_fault(uint32_t msp, uint32_t psp, uint32_t exc_return,
 
 #ifdef CONFIG_DEBUG_COREDUMP
 	z_arm_coredump_fault_sp = POINTER_TO_UINT(esf);
-#endif
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+	/* Gdb expects a stack pointer that does not include the exception stack frame in order to
+	 * unwind. So adjust the stack pointer accordingly.
+	 */
+	z_arm_coredump_fault_sp += sizeof(esf->basic);
```

Comment:
Doesn't this also apply to Cortex-M0/+ and Cortex-M23 (Armv6-m and Armv8-m.baseline) ?

Reply:
Yes, looks like you are right, I'll update